core/src/main/scala/filodb.core/store/IngestionConfig.scala

```diff
@@ -127,17 +128,19 @@ object IngestionConfig {
     * Allows the caller to decide what to do with configuration parsing errors and when.
     * Fails if no dataset is provided by the config submitter.
     */
-  private[core] def apply(sourceConfig: Config): Try[IngestionConfig] =
+  private[core] def apply(topLevelConfig: Config): Try[IngestionConfig] = {
     for {
-      resolved  <- sourceConfig.resolveT
-      dataset   <- resolved.stringT(DatasetRefKey) // fail fast if missing
-      factory   <- resolved.stringT(SourceFactory) // fail fast if missing
-      numShards <- numShards(resolved)             // fail fast if missing
-      minNodes  <- minNumNodes(resolved)           // fail fast if missing
-      streamConfig = resolved.as[Option[Config]](IngestionKeys.SourceConfig).getOrElse(ConfigFactory.empty)
-      ref          = DatasetRef.fromDotString(dataset)
-      storeConf <- streamConfig.configT("store").map(StoreConfig.apply)
-    } yield IngestionConfig(ref, resolved, factory, streamConfig, storeConf)
+      resolved <- topLevelConfig.resolveT
```

Comment:
Can you please line up the <- again?  It really looks much nicer with it like that.